mlir/lib/Analysis/Presburger/Coalesce.cpp

```diff
@@ -0,0 +1,686 @@
+#include "mlir/Analysis/Presburger/Coalesce.h"
+#include "mlir/Analysis/AffineStructures.h"
+#include "mlir/Analysis/Presburger/Simplex.h"
+#include "mlir/Analysis/Presburger/Set.h"
+#include <iostream>
+
+using namespace mlir;
+// struct for classified constraints
+struct Info {
+  SmallVector<SmallVector<int64_t, 8>, 8> redundant;
```

Comment:
Same as before, I'd prefer to make those read only, when the functionality is fully implemented.

Reply:
This will not prevent you from modifying them and introducing unwanted side effects. I would never expect a function that uses an Info object to modify it.